indexer/search_index_builder.cpp

```diff
@@ -263,171 +256,96 @@ class FeatureInserter
 {
   Platform & platform = GetPlatform();
 
-  FilesContainerR readContainer(platform.GetReader(fName));
-  if (readContainer.IsExist(COMPRESSED_SEARCH_INDEX_FILE_TAG) && !forceRebuild)
+  FilesContainerR readContainer(platform.GetReader(filename));
+  if (readContainer.IsExist(SEARCH_INDEX_FILE_TAG) && !forceRebuild)
     return true;
 
-  string const indexFile = platform.WritablePathForFile("compressed-search-index.tmp");
-  MY_SCOPE_GUARD(indexFileGuard, bind(&FileWriter::DeleteFileX, indexFile));
+  string mwmName = filename;
+  my::GetNameFromFullPath(mwmName);
+  my::GetNameWithoutExt(mwmName);
+  string const indexFilePath = platform.WritablePathForFile(mwmName + ".sdx.tmp");
+  MY_SCOPE_GUARD(indexFileGuard, bind(&FileWriter::DeleteFileX, indexFilePath));
+  string const stringsFilePath = platform.WritablePathForFile(mwmName + ".sdx.strings.tmp");
+  MY_SCOPE_GUARD(stringsFileGuard, bind(&FileWriter::DeleteFileX, stringsFilePath));
 
   try
   {
     {
-      FileWriter indexWriter(indexFile);
-      BuildCompressedSearchIndex(readContainer, indexWriter);
+      FileWriter indexWriter(indexFilePath);
+      BuildSearchIndex(readContainer, indexWriter, stringsFilePath);
+      LOG(LINFO, ("Search index size = ", indexWriter.Size()));
     }
     {
       FilesContainerW writeContainer(readContainer.GetFileName(), FileWriter::OP_WRITE_EXISTING);
-      FileWriter writer = writeContainer.GetWriter(COMPRESSED_SEARCH_INDEX_FILE_TAG);
-      rw_ops::Reverse(FileReader(indexFile), writer);
+      FileWriter writer = writeContainer.GetWriter(SEARCH_INDEX_FILE_TAG);
+      rw_ops::Reverse(FileReader(indexFilePath), writer);
     }
   }
   catch (Reader::Exception const & e)
   {
-    LOG(LERROR, ("Error while reading file: ", e.Msg()));
+    LOG(LERROR, ("Error while reading file:", e.Msg()));
     return false;
   }
   catch (Writer::Exception const & e)
   {
-    LOG(LERROR, ("Error writing index file: ", e.Msg()));
+    LOG(LERROR, ("Error writing index file:", e.Msg()));
     return false;
   }
 
   return true;
 }
 
-void BuildCompressedSearchIndex(FilesContainerR & container, Writer & indexWriter)
+void BuildSearchIndex(FilesContainerR & container, Writer & indexWriter,
+                      string const & stringsFilePath)
 {
+  using TValue = FeatureIndexValue;
+
   Platform & platform = GetPlatform();
 
-  LOG(LINFO, ("Start building compressed search index for", container.GetFileName()));
+  LOG(LINFO, ("Start building search index for", container.GetFileName()));
   my::Timer timer;
 
-  string stringsFilePath = platform.WritablePathForFile("strings.tmp");
-  StringsFile<FeatureIndexValue> stringsFile(stringsFilePath);
-  MY_SCOPE_GUARD(stringsFileGuard, bind(&FileWriter::DeleteFileX, stringsFilePath));
-
   CategoriesHolder categoriesHolder(platform.GetReader(SEARCH_CATEGORIES_FILE_NAME));
 
-  AddFeatureNameIndexPairs(container, categoriesHolder, stringsFile);
+  FeaturesVectorTest features(container);
```

Comment:
Yet the documentation for FeatureVectorTest explicitly allows it.
I can rename it to FeaturesVectorStandAlone or something, though not in this PR.
@vng what do you think?